### PR TITLE
mayhem integration

### PR DESCRIPTION
**Pull request by evanrichter**

### CODE REVIEW

File: fuzz/fuzz_targets/fuzz_encode_reconstruct.rs [hunk 1]
```diff
@@ -2,7 +2,7 @@
 #[macro_use] extern crate libfuzzer_sys;
```
Comment (xansec) [inline fuzz/fuzz_targets/fuzz_encode_reconstruct.rs:2]: Can you comment on these changes?

File: fuzz/fuzz_targets/fuzz_encode_verify.rs [hunk 2]
```diff
@@ -2,7 +2,7 @@
 #[macro_use] extern crate libfuzzer_sys;
```
Comment (xansec) [inline fuzz/fuzz_targets/fuzz_encode_verify.rs:2]: Can you comment on these changes?

**Review by xansec — CHANGES_REQUESTED** [review]
Please make the requested changes.

File: fuzz/fuzz_targets/fuzz_encode_reconstruct.rs (hunk 1 above)
Comment (evanrichter) [reply fuzz/fuzz_targets/fuzz_encode_reconstruct.rs:2]: their fuzzer wasn't compiling due to api changes, and there were only 2 choices of types and functions to choose, and only one of those compiled and ran without errors for sufficient time.

File: fuzz/fuzz_targets/fuzz_encode_verify.rs (hunk 2 above)
Comment (evanrichter) [reply fuzz/fuzz_targets/fuzz_encode_verify.rs:2]: same as above

**Comment by xansec** [comment]
Approved!

**Pull request merged**